### DISCUSSION TITLE
Should I versioning `tsbuildinfo`?

### DISCUSSION THREAD

**Discussion by raulfdm** [vercel/next.js#30815] [Help] [answered]
Hey peps.

I notice that after migrating to NextJS 12, my TS Config got a new config called `incremental` and that generates a `tsconfig.tsbuildinfo` files at root level.

If I understood correctly from TypeScript docs, it's meant to make the compilation faster and can be deleted safely.

In that sense, we don't have to version it, right? I was wondering if we track that the production build would be even faster but I'm not sure.

Any ways, silly question but I couldn't find any resources on that (for NextJS ofc).

### Resources
- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#faster-subsequent-builds-with-the---incremental-flag

**Comment by monsieurnebo** [comment]
Same question over here 👋 

I untracked it based on the same "can be deleted safely" you found on the doc, but I would like a confirmation about the versioning part 🤔

**Comment by wwwdepot** [comment, marked as answer]
Next.js generates its own `.tsbuildinfo` at `.next/cache`. So, I guess you don't commit `tsconfig.tsbuildinfo` to version control.

**Reply by raulfdm** [reply]
Makes sense. Thanks for this input!
